tensorrt_llm/_torch/pyexecutor/connector.py

```diff
@@ -0,0 +1,452 @@
+# SPDX-FileCopyrightText: Copyright (c) 2022-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the primary interface for the KV Cache Connector.
+
+The KV Cache Connector is a component that allows for remote KV cache access.
+It is responsible for:
+- Orchestrating the loading and saving of KV cache blocks.
+- Managing asynchronous block tx/rx.
+
+It can be used to provide functionalities such as:
+1. Disagg
+2. KV offload/onboard
+3. KV cache sharing
+4. P2P KV cache transfer
+etc.
+
+The Connector API is split into two parts:
+1. The scheduler, which is responsible for orchestration, and building metadata for the workers.
+2. The worker, which performs and monitors transfers indicated by the scheduler's metadata.
+
+To implement a custom KV connector, you need to implement both the scheduler and worker-side interfaces.
+"""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from typing import Any, Callable, Optional
+
+import torch
+
+from tensorrt_llm._utils import mpi_allgather, mpi_broadcast, mpi_rank
+from tensorrt_llm.bindings import LlmRequestState
+from tensorrt_llm.bindings.executor import ExecutorConfig
+from tensorrt_llm.bindings.internal.batch_manager import \
+    KvCacheConnectorManager as KvCacheConnectorManagerCpp
+from tensorrt_llm.bindings.internal.batch_manager import LlmRequest
+
+from .scheduler import ScheduledRequests
+
+
+# Used to store data for a single inflight request.
+@dataclass
+class RequestData:
+    # The request ID.
+    request_id: int
+    # The new tokens that were generated in the prior forward pass.
+    new_tokens: list[int]
+    # The new block IDs allocated in the prior forward pass.
+    new_block_ids: list[int]
```

Comment:
```suggestion
    new_tokens: List[int]
    # The new block IDs allocated in the prior forward pass.
    new_block_ids: List[int]
```
to be consistent with the rest of the codebase.